app/models/user.rb

```diff
@@ -1,6 +1,43 @@
 class User < ApplicationRecord
-  include Clearance::User
-  has_one :profile, dependent: :destroy
+  scope :visible_for_user, ->(user) { where(visible: true).or(where(id: user&.id)) }
+  scope :members, -> { where('joined_at IS NOT NULL AND left_at is NULL') }
 
-  validates :email, uniqueness: { case_sensitive: false }
+  validates :email,
+    uniqueness: { case_sensitive: false },
+    email_format: true
+  validates :preferred_name, presence: true
+  validates :full_name, presence: true
+
+  before_save do
+    self.email_confirmed = false if email_changed?
+  end
+
+  has_secure_password
+  has_secure_token
```

Comment:
I was wondering about the uniqueness of these tokens. The `generate_unique_secure_token` method in that `SecureToken` module (where `has_secure_token` comes from) is not _guaranteed_ unique... but that's not a problem because the chances of a collision are one in two tredecillion. (1 in 2,100,254,120,907,352,485,526,230,505,830,591,911,428,096)

That should be unique enough for us :)